File: pset1/exercises/lab1/exercises/credit.c
#include <stdio.h>
#include <cs50.h>

void validateCard(long n);

int main(void)
{
    long card = get_long("Number of card: ");
    validateCard(card);
}

bool countNumbers(long n)
{
    long number = n;
    long count = 0;
    do
    {
        number = number / 10;
        count++;
    }while(number != 0);
    if(count == 13 || count == 15 || count == 16)
    {
        return true;
    }else
    {
        return false;
    }
}

string typeCard(long n)
{
    if(n == 34 || n == 37)
    {
        return "AMEX\n";
    }else if(n >= 51 && n <= 55)
    {
        return "MASTERCARD\n";
    }else if(n >= 40 && n <= 49)
    {
        return "VISA\n";
    }else
    {
        return "INVALID\n";
    }
}

void validateCard(long n)
{
    long number = n;
    int sum = 0;
    string type = "INVALID\n";
    if(countNumbers(number))
    {
        bool in = 1;
        bool dou = 0;
        do
        {
            if(in && number > 10)
            {
                sum += number % 10;
                number = (long)number / 10;
                in = 0;
                dou = 1;
            }
            if(number < 100 && number > 10)
            {
                type = typeCard(number);
            }
            if(dou && number > 10)
            {
                int sum1 = (number % 10) * 2;
                if(sum1 > 10)
                {
                    sum += sum1 % 10;
                    sum += (long)sum1 / 10;
                }else
                {
                    sum += sum1;
                }
                number = number / 10;
                in = 1;
                dou = 0;
            }
            if(number < 100 && number > 10)
            {
                type = typeCard(number);
            }
        }while(number > 10);
        if(in)
        {
            sum += number;
            int total = sum % 10;
            if(total == 0)
            {
                printf("%s", type);
            }else
            {
                printf("INVALID\n");
            }
        }
        if(dou)
        {
            sum += number * 2;
            int total = sum % 10;
            if(total == 0)
            {
                printf("%s", type);
            }else
            {
                printf("INVALID\n");
            }
        }
    }else
    {
        printf("INVALID\n");
    }
}

File: pset6/exercises/credit.py
from cs50 import get_int

def countNumbers(n):
    count = 0
    while n != 0:
        n = int(n / 10)
        count += 1
    if count == 13 or count == 15 or count == 16:
        return True
    else:
        return False

def typeCard(n):
    if n == 34 or n == 37:
        return "AMEX"
    elif n >= 51 and n <= 55:
        return "MASTERCARD"
    elif n >= 40 and n <= 49:
        return "VISA"
    else:
        return "INVALID"

def validateCard(n):
    sum = 0
    type = "INVALID"
    if countNumbers(n):
        first = True
        secund = False
        while n > 10:
            if first and n > 10:
                sum += n % 10
                n = int(n / 10)
                first = False
                secund = True
            if n < 100 and n > 10:
                type = typeCard(n)
            if n < 10:
                break
            if secund and n > 10:
                sum1 = (n % 10) * 2
                if sum1 > 10:
                    sum += sum1 % 10
                    sum += int(sum1 / 10)
                else:
                    sum += sum1
                n = int(n / 10)
                first = True
                secund = False
            if n < 100 and n > 10:
                type = typeCard(n)
            if n < 10:
                break
        if first:
            sum += n % 10
            if (sum % 10) == 0:
                print(type)
            else:
                print("INVALID")
        if secund:
            sum1 = (n % 10) * 2
            if sum1 > 10:
                sum += sum1 % 10
                sum += sum1 / 10
            else:
                sum += sum1
            if (sum % 10) == 0:
                print(type)
            else:
                print("INVALID")
    else:
        print("INVALID")

number = get_int("Number of card: ")
validateCard(number)